lib/unleash/feature_toggle.rb

```diff
@@ -97,10 +97,10 @@ def sum_variant_defs_weights
     end
 
     def variant_salt(context, stickiness = "default")
-      return context.get_by_name(stickiness) unless stickiness == "default"
-      return context.user_id unless context.user_id.to_s.empty?
-      return context.session_id unless context.session_id.to_s.empty?
-      return context.remote_address unless context.remote_address.to_s.empty?
+      return context.get_by_name(stickiness) unless stickiness == "default" || context.nil?
+      return context.user_id unless context&.user_id.to_s.empty?
+      return context.session_id unless context&.session_id.to_s.empty?
+      return context.remote_address unless context&.remote_address.to_s.empty?
```

Comment:
Do you mean tests that pass in a context missing one of `user_id, session_id, remote_address` additively each time?

Reply:
I am working on some extra tests in FeatureToggle, so it can make sense to call it out of scope for the PR.

 But the fix in `feature_toggle.rb` that I mention could be addressed here right away. FYI: I found the issue when writing these extra tests locally.